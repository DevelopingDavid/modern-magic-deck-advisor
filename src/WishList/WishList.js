import React from 'react';
import WishListItem from '../WishListItem/WishListItem.js'
import '../styles/main.scss';

function WishList(props) {
  return(
    <div>
      <h1>Wish List:</h1>
      <ul>
        {
          props.wishList.map((card, index) => {
            return(
              <WishListItem
                cardName={card.cardName}
                cardIndex={index}
                wishListCount={card.wishListCount}
                key={card.cardName}
                wishList={props.wishList}
                saveArray={props.saveArray}/>
            )
          })
        }
      </ul>
    </div>
  )
}

export default WishList;