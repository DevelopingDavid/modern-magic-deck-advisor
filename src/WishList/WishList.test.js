import React from 'react';
import WishList from './WishList';
import { shallow } from 'enzyme';
import { isMainThread } from 'worker_threads';

const wishList = [{ cardName: "Blood Moon" }, { cardName: "Mox Opal" }, { cardName: "Breeding Pool" }];
const removeWishListItemMock = jest.fn();

describe('WishList', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(
      <WishList
        wishList={wishList}
        removeWishListItem={removeWishListItemMock}/>
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})