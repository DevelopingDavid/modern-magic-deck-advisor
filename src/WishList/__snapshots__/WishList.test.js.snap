// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WishList should match the snapshot 1`] = `
<div>
  <h1
    className="list--h1"
  >
    Wish List:
  </h1>
  <ul
    className="list--ul"
  >
    <WishListItem
      cardIndex={0}
      cardName="Blood Moon"
      key="Blood Moon"
      wishList={
        Array [
          Object {
            "cardName": "Blood Moon",
          },
          Object {
            "cardName": "Mox Opal",
          },
          Object {
            "cardName": "Breeding Pool",
          },
        ]
      }
    />
    <WishListItem
      cardIndex={1}
      cardName="Mox Opal"
      key="Mox Opal"
      wishList={
        Array [
          Object {
            "cardName": "Blood Moon",
          },
          Object {
            "cardName": "Mox Opal",
          },
          Object {
            "cardName": "Breeding Pool",
          },
        ]
      }
    />
    <WishListItem
      cardIndex={2}
      cardName="Breeding Pool"
      key="Breeding Pool"
      wishList={
        Array [
          Object {
            "cardName": "Blood Moon",
          },
          Object {
            "cardName": "Mox Opal",
          },
          Object {
            "cardName": "Breeding Pool",
          },
        ]
      }
    />
  </ul>
</div>
`;
